packages/backend/src/managers/playground/McpServerManager.spec.ts

```diff
@@ -16,95 +16,106 @@
 import path from 'node:path';
-import { type McpServer, McpServerType } from '../../models/mcpTypes';
+import { type RpcExtension } from '@shared/messages/MessageProxy';
+import { type McpClient, type McpServer, McpServerType, type McpSettings } from '@shared/models/McpSettings';
 import { McpServerManager } from './McpServerManager';
+import { JsonWatcher } from '../../utils/JsonWatcher';
+import { toMcpClients } from '../../utils/mcpUtils';
 
-vi.mock('node:fs');
-vi.mock('node:fs/promises');
+vi.mock('../../utils/JsonWatcher');
+vi.mock('../../utils/mcpUtils');
 
-/* eslint-disable sonarjs/no-nested-functions */
-describe('McpServerManager', () => {
-  let appUserDirectory: string;
-  beforeEach(async () => {
-    vi.resetAllMocks();
-    appUserDirectory = path.join('/', 'tmp', 'mcp-server-manager-test-');
-  });
-  describe('load', () => {
-    test('with no file, returns empty array', async () => {
-      const mcpServerManager = new McpServerManager(appUserDirectory);
-      const mcpServers = await mcpServerManager.load();
-      expect(mcpServers).toEqual([]);
-    });
-    test('with empty file, returns empty array', async () => {
-      const mcpServerManager = new McpServerManager(appUserDirectory);
-      vi.mocked(promises.readFile).mockResolvedValue('{}');
-      const mcpServers = await mcpServerManager.load();
-      expect(mcpServers).toEqual([]);
-    });
-    test('with invalid JSON, returns empty array', async () => {
-      const mcpServerManager = new McpServerManager(appUserDirectory);
-      vi.mocked(promises.readFile).mockResolvedValue('{invalid json}');
-      const mcpServers = await mcpServerManager.load();
-      expect(mcpServers).toEqual([]);
-    });
-    describe('with valid JSON', () => {
-      let mcpServers: McpServer[];
-      beforeEach(async () => {
-        const mcpSettings = {
-          servers: {
-            'stdio-ok': {
-              enabled: true,
-              type: 'stdio',
-              command: 'npx',
-              args: ['-y', 'kubernetes-mcp-server'],
-            },
-            'sse-ok': {
-              enabled: true,
-              type: 'sse',
-              url: 'https://echo.example.com/sse',
-              headers: {
-                foo: 'bar',
-              },
-            },
-            'invalid-type': {
-              enabled: true,
-              type: 'invalid',
-              url: 'https://echo.example.com/sse',
-            },
+let mockJsonWatcher: JsonWatcher<McpSettings>;
+let rpcExtension: RpcExtension;
+let update: (mcpSettings: McpSettings) => void;
+let appUserDirectory: string;
+let mcpServerManager: McpServerManager;
+beforeEach(async () => {
+  vi.resetAllMocks();
+  mockJsonWatcher = {
+    init: vi.fn(),
+    dispose: vi.fn(),
+    onContentUpdated: vi.fn((fn: (mcpSettings: McpSettings) => void) => (update = fn)),
+  } as unknown as JsonWatcher<McpSettings>;
+  vi.mocked(JsonWatcher).mockImplementation(() => mockJsonWatcher);
```

Comment:
```suggestion
  vi.mocked(JsonWatcher).mockReturnValue(mockJsonWatcher);
```

Here, when mocking a constructor, you may use mock return value directly